storage/remote/client.go

```diff
@@ -136,8 +138,15 @@ func NewWriteClient(name string, conf *ClientConfig) (WriteClient, error) {
 	if err != nil {
 		return nil, err
 	}
-
 	t := httpClient.Transport
+
+	if conf.SigV4Config.Enabled {
```

Comment:
I don't do a similar check for `NewReadClient` which doesn't support this, but maybe it should check and panic if it's set?

Reply:
We could enable it for remote read as well, even if not useful in pratice at the moment.

Reply:
Wait, NewReadClient does not have the Sivg4 enable field so there is no issue at all. This is 2 different structs.

Reply:
They both use `ClientConfig`, which is what I'm mostly concerned about here.

Reply:
But the field is for RemoteWriteConfig, in read config you can not have sigv4.